src/components/surveys/survey-design/question-edit/SelectPhoneBottomMenu.tsx

```diff
@@ -227,15 +232,33 @@ const SelectPhoneBottomMenu: FunctionComponent<{
     }
   }
 
+  const isOptionSupported = (optionKey: SelectorOptionType): boolean => {
+    if (step.singleChoice ?? true) {
+      // If this is a single choice step then *only* other is supported 
+      return optionKey === 'OTHER' && step.baseType === 'string'
+    } else {
+      // If this is multiple answer then everything *but* Other is supported
+      return optionKey !== 'OTHER'
+    }
+  }
+
   const getOptions = (): SelectorOptionType[] => {
-    const keyArray = Array.from(OPTIONS.keys())
-    //for single choice only add other. If it's an integer value 'other' is disabled
-    if (step.singleChoice) {
-      const result = keyArray.filter(optionKey => optionKey === 'OTHER')
+    return Array.from(OPTIONS.keys()).filter(optionKey => isOptionSupported(optionKey) && !isOptionAlreadyAdded(optionKey))
+  }
 
-      return result
-    }
-    return keyArray
+  const SpecialOptionsMenuItems: FunctionComponent = () => {
+    let options = getOptions()
+    return options.length === 0 ? <NoMenuItemOptions /> :
+    <>
+      {options.map(optionKey => (
+        <StyledMenuItem
+          key={optionKey}
+          disabled={isOptionAlreadyAdded(optionKey)}
+          onClick={() => addGenericResponse(optionKey)}>
+          {OPTIONS.get(optionKey)?.selectDisplayLabel}
+        </StyledMenuItem>
+      ))}
+    </>
   }
```

Comment:
Nice! Would it be possible to move `SpecialOptionsMenuItems` out of the parent component (`SelectPhoneBottomMenu`), then pass in props (e.g. `options`), so that `SpecialOptionsMenuItems` is reusable?

Reply:
I'm sure it would be possible, but I couldn't quite figure out how to pass in the properties without it throwing errors. Also the logic for this is tied to the model object specific to the menu items in the ":" menu so it made sense (to me) to only define it internally to the `SelectPhoneBottomMenu`.